### ISSUE TITLE
Missing types

### ISSUE THREAD

**Issue by cfbraun** [soypat/ether-swtch#1]
Hello,

the code uses drivers.Datagrammer and drivers.Packet. These types don't seem to exist in the current version of tinygo/x/drivers. Can you tell me where to find these types?

**Comment by soypat** [comment]
Hey! I was using my [enc28j60 branch on my drivers repo](https://github.com/soypat/drivers/blob/enc28j60/datagrammer.go) (link to the file with the types). To set up a development environment I'd suggest cloning this branch and using a go module replace directive of the tinygo drivers package with the enc28j60 branch on my drivers repo.

**A few things to note**:

1. This library will be deprecated. I'm keeping it alive until I or someone else gets around to [writing a better version](https://github.com/tinygo-org/tinygo/issues/3028).
2. This library may have bugs
3. This library is very poorly written with hard to follow control flow

That said, I hope it works for you!

**Comment by cfbraun** [comment]
Hello, thank’s for your reply! I have used both UIP and LwIP for different embedded projects in the past, but when using Go I would like to avoid linking (and maybe back-linking if HW support is implemented in Go) C libraries. So implementing (the necessary subset of) a IP stack in TinyGo looks promising. I am aware of the limitations of your library, but it looks like a very good starting point anyway.

Thanks again!
Am 24. Sept. 2022, 19:42 +0200 schrieb Patricio Whittingslow ***@***.***>:

Hey! I was using my enc28j60 branch on my drivers repo<https://github.com/soypat/drivers/blob/enc28j60/datagrammer.go> (link to the file with the types). To set up a development environment I'd suggest cloning this branch and using a go module replace directive of the tinygo drivers package with the enc28j60 branch on my drivers repo.

A few things to note:

  1.  This library will be deprecated. I'm keeping it alive until I or someone else gets around to writing a better version<https://github.com/tinygo-org/tinygo/issues/3028>.
  2.  This library may have bugs that I will not fix
  3.  This library is very poorly written with hard to follow control flow

That said, I hope it works for you!

**Comment by soypat** [comment]
@cfbraun Hey! just a friendly heads up I am archiving this repository and closing this issue.

Development of a embedded lwip-like stack for embedded systems for Go has taken off at https://github.com/soypat/seqs, so probably check that out if you are still interested in this line of developments.
